### PR TITLE
Postgres + docker-compose

### PR DESCRIPTION
**Pull request by vsalavatov**

### CODE REVIEW

File: internal/interface/repository/postgres/articles.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	return &ArticleRepo{db: db}
+}
+
+func (a *ArticleRepo) ArticleById(id model.ArticleId) (model.Article, error) {
+	rows, err := a.db.Query("SELECT Id, Title, Abstract, LastUpdateTimestamp FROM Articles WHERE Id = $1;", id)
+	if err != nil {
+		return model.Article{}, err
+	}
+	defer rows.Close()
+	for rows.Next() {
```
Comment (M0nsteR) [inline internal/interface/repository/postgres/articles.go:28]: This needs refactoring, in my view. Consider using fewer SQL queries to extract Article -> Authors data. Keep in mind the high cognitive load of deeply nested code.

File: internal/interface/repository/implicitrepos/updates_through_queries.go [hunk 2]
```diff
@@ -84,7 +81,7 @@ func (u *UpdatesRepoThroughQueries) GetSearchSubscriptionsUpdates(id model.UserI
 				lastArticleUpdateTimestamp = articleMeta.LastUpdateTimestamp
 			}
 		}
-		lastAccessTimestamp, err := u.historyRepo.GetSearchLastAccessTimestamp(id, query)
+		lastAccessTimestamp, err := u.searchSubsRepo.GetSearchLastAccessTimestamp(id, query)
 		if err != nil && err != domain.NeverAccessed {
```
Comment (M0nsteR) [inline internal/interface/repository/implicitrepos/updates_through_queries.go]: Make it more readable. Consider using early returns: https://medium.com/swlh/return-early-pattern-3d18a41bba8

File: internal/interface/crawler/crawler.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+func (c *Crawler) responseProcessor(parseErr *error, urlQueue *[]string, cfg *Configuration) func(response *colly.Response) {
+	return func(response *colly.Response) {
+		originalUrl := response.Request.URL.String()
+		dom, err := goquery.NewDocumentFromReader(bytes.NewReader(response.Body))
+		if err != nil {
+			parseErr = &err
+			return
+		}
+		dom.Find("a[href]").Each(func(i int, s *goquery.Selection) {
```
Comment (M0nsteR) [inline internal/interface/crawler/crawler.go]: Time to split this func.

File: internal/interface/crawler/crawler.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	defer rows.Close()
+	for rows.Next() {
+		var cnt int
+		err = rows.Scan(&cnt)
+		return cnt, err
+	}
+	return 0, fmt.Errorf("unexpected query result")
+}
+
+func (c *Crawler) upsertArticle(article model.Article) (bool, error) {
```
Comment (M0nsteR) [inline internal/interface/crawler/crawler.go:69]: I failed miserably to wrap my head around this logic..

File: internal/interface/crawler/crawler.go [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+	for rows.Next() {
+		cfg := Configuration{}
+		err := rows.Scan(&cfg.Id, &cfg.DesiredArticleCount, &cfg.RootURL)
+		return cfg, err
+	}
+	return Configuration{}, ErrNoConfigs
+}
+
+func (c *Crawler) getArticlesCount() (int, error) {
+	rows, err := c.db.Query("SELECT COUNT(*) FROM Articles;")
```
Comment (M0nsteR) [inline internal/interface/crawler/crawler.go]: You do understand this is a slow aggregate query? Some sort of in-memory cached value might be in order.

File: initdb.sql [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+CREATE TABLE IF NOT EXISTS AccountSearchRelations (
+    UserId integer REFERENCES Accounts (Id),
+    Search text,
+    IsSubscribed boolean,
+    LastAccess bigint
+);
+
+CREATE TABLE IF NOT EXISTS CrawlerConfig (
+    Id integer primary key,
```
Comment (M0nsteR) [inline initdb.sql]: Why do you need a primary key here? The RootURL field is already unique.

**Review by M0nsteR — CHANGES_REQUESTED** [review]
Too many lines of code for a meaningful code review..

File: internal/interface/repository/implicitrepos/updates_through_queries.go (hunk 2 above)
Comment (vsalavatov) [reply internal/interface/repository/implicitrepos/updates_through_queries.go]: fixed

File: internal/interface/crawler/crawler.go (hunk 3 above)
Comment (vsalavatov) [reply internal/interface/crawler/crawler.go]: fixed

File: internal/interface/crawler/crawler.go (hunk 4 above)
Comment (vsalavatov) [reply internal/interface/crawler/crawler.go:69]: made it more readable

File: internal/interface/crawler/crawler.go (hunk 5 above)
Comment (vsalavatov) [reply internal/interface/crawler/crawler.go]: fixed

File: initdb.sql (hunk 6 above)
Comment (vsalavatov) [reply initdb.sql]: fixed

**Review by M0nsteR — APPROVED** [review]
Nice and fast, good to go.

**Pull request merged**